docs/api.rst

```diff
@@ -9,97 +9,97 @@ better guide to the intended usage.
 Interfaces
 ==========
 
-.. autoclass:: zope.schema.interfaces.IField
-.. autoclass:: zope.schema.interfaces.IFromUnicode
-.. autoclass:: zope.schema.interfaces.IChoice
-.. autoclass:: zope.schema.interfaces.IContextAwareDefaultFactory
-.. autoclass:: zope.schema.interfaces.IOrderable
-.. autoclass:: zope.schema.interfaces.ILen
-.. autoclass:: zope.schema.interfaces.IMinMax
-.. autoclass:: zope.schema.interfaces.IMinMaxLen
-.. autoclass:: zope.schema.interfaces.IInterfaceField
-.. autoclass:: zope.schema.interfaces.IBool
-.. autoclass:: zope.schema.interfaces.IObject
-.. autoclass:: zope.schema.interfaces.IDict
+.. autointerface:: zope.schema.interfaces.IField
+.. autointerface:: zope.schema.interfaces.IFromUnicode
+.. autointerface:: zope.schema.interfaces.IChoice
+.. autointerface:: zope.schema.interfaces.IContextAwareDefaultFactory
+.. autointerface:: zope.schema.interfaces.IOrderable
+.. autointerface:: zope.schema.interfaces.ILen
+.. autointerface:: zope.schema.interfaces.IMinMax
+.. autointerface:: zope.schema.interfaces.IMinMaxLen
+.. autointerface:: zope.schema.interfaces.IInterfaceField
+.. autointerface:: zope.schema.interfaces.IBool
+.. autointerface:: zope.schema.interfaces.IObject
+.. autointerface:: zope.schema.interfaces.IDict
```

Comment:
Weird, I would've expected these changes to be added in the commit which added `repoze.sphinx.autointerface` to the Sphinx extensions list.

Reply:
That has an interesting history, I'm glad you brought it up :) `repoze.sphinx.autointerface` was installed on RTD [in the original RTD commit](https://github.com/zopefoundation/zope.schema/commit/fb35d50b390ea9f084e6262a570c5ad219cfa118) four years ago. Two years after that, *I* added `api.rst` [with the `.. autoclass::` directives](https://github.com/zopefoundation/zope.schema/commit/fb60445848dd821d05cf74168a788ae39fdd0b0e). A year after that, `r.s.a` got [added to setup.py](https://github.com/zopefoundation/zope.schema/commit/31a3878946454b4f62d322ebf88187e6b44af460) (which I think is when we started running sphinx on Travis CI and in tox). 

I tested as far back as Sphinx 1.4, and all versions emit copious warnings like ["WARNING: error while formatting arguments for zope.schema.interfaces.IField: 'InterfaceClass' object has no attribute '__mro__'"](https://travis-ci.org/zopefoundation/zope.schema/jobs/425444010#L543)...but only on Python 3. 

So my best guess is that I didn't locally build the docs with Python 3 before this, and because everything on CI was green, nobody noticed that [all the interface bodies are blank or RTD](https://zopeschema.readthedocs.io/en/latest/api.html#interfaces).